aws/credentials/processcreds/provider_test.go

```diff
@@ -0,0 +1,530 @@
+		t.Errorf("error getting credentials: %v", err)
+	}
+
+	if e, a := "notFromCredProcAccess", creds.AccessKeyID; e != a {
+		t.Errorf("expected %v, got %v", e, a)
+	}
+
+	if e, a := "notFromCredProcSecret", creds.SecretAccessKey; e != a {
+		t.Errorf("expected %v, got %v", e, a)
+	}
+
+}
+
+func TestProcessProviderBadCommand(t *testing.T) {
+	oldEnv := preserveImportantStashEnv()
+	defer awstesting.PopEnv(oldEnv)
+
+	creds := processcreds.NewCredentials("/bad/process")
+	_, err := creds.Get()
+	if err.(awserr.Error).Code() != processcreds.ErrCodeProcessProviderExecution {
+		t.Errorf("expected %v, got %v", processcreds.ErrCodeProcessProviderExecution, err)
+	}
+}
+
+func TestProcessProviderMoreEmptyCommands(t *testing.T) {
+	oldEnv := preserveImportantStashEnv()
+	defer awstesting.PopEnv(oldEnv)
+
+	creds := processcreds.NewCredentials("")
+	_, err := creds.Get()
+	if err.(awserr.Error).Code() != processcreds.ErrCodeProcessProviderExecution {
+		t.Errorf("expected %v, got %v", processcreds.ErrCodeProcessProviderExecution, err)
+	}
+
+}
+
+func TestProcessProviderExpectErrors(t *testing.T) {
+	oldEnv := preserveImportantStashEnv()
+	defer awstesting.PopEnv(oldEnv)
+
+	creds := processcreds.NewCredentials(
+		fmt.Sprintf(
+			"%s %s",
+			getOSCat(),
+			strings.Join(
+				[]string{"testdata", "malformed.json"},
+				string(os.PathSeparator))))
+	_, err := creds.Get()
+	if err.(awserr.Error).Code() != processcreds.ErrCodeProcessProviderParse {
+		t.Errorf("expected %v, got %v", processcreds.ErrCodeProcessProviderParse, err)
+	}
+
+	creds = processcreds.NewCredentials(
+		fmt.Sprintf("%s %s",
+			getOSCat(),
+			strings.Join(
+				[]string{"testdata", "wrongversion.json"},
+				string(os.PathSeparator))))
+	_, err = creds.Get()
+	if err.(awserr.Error).Code() != processcreds.ErrCodeProcessProviderVersion {
+		t.Errorf("expected %v, got %v", processcreds.ErrCodeProcessProviderVersion, err)
+	}
+
+	creds = processcreds.NewCredentials(
+		fmt.Sprintf(
+			"%s %s",
+			getOSCat(),
+			strings.Join(
+				[]string{"testdata", "missingkey.json"},
+				string(os.PathSeparator))))
+	_, err = creds.Get()
+	if err.(awserr.Error).Code() != processcreds.ErrCodeProcessProviderRequired {
+		t.Errorf("expected %v, got %v", processcreds.ErrCodeProcessProviderRequired, err)
+	}
+
+	creds = processcreds.NewCredentials(
+		fmt.Sprintf(
+			"%s %s",
+			getOSCat(),
+			strings.Join(
+				[]string{"testdata", "missingsecret.json"},
+				string(os.PathSeparator))))
+	_, err = creds.Get()
+	if err.(awserr.Error).Code() != processcreds.ErrCodeProcessProviderRequired {
+		t.Errorf("expected %v, got %v", processcreds.ErrCodeProcessProviderRequired, err)
+	}
+
+}
+
+func TestProcessProviderTimeout(t *testing.T) {
+	oldEnv := preserveImportantStashEnv()
+	defer awstesting.PopEnv(oldEnv)
+
+	command := "/bin/sleep 2"
+	if runtime.GOOS == "windows" {
+		// "timeout" command does not work due to pipe redirection
+		command = "C:\\Windows\\system32\\ping -n 2 127.0.0.1>nul"
```

Comment:
system32 ist required to be on the `C:\` drive, nor is the win dir required to be "Windows". :)  

Should use `SYSTEM32` [env var](https://docs.microsoft.com/en-us/windows/deployment/usmt/usmt-recognized-environment-variables) instead for windows.

Reply:
I have fixed this (preserved system32 env when popping env).